torchtext/data/dataset.py

```diff
@@ -323,8 +323,22 @@ def stratify(examples, strata_field):
 
 
 def rationed_split(examples, train_ratio, test_ratio, val_ratio, rnd):
-    # Create a random permutation of examples, then split them
-    # by ratio x length slices for each of the train/test/dev? splits
+    """Create a random permutation of examples, then split them by ratios
```

Comment:
This looks a lot like [`torch.utils.data.random_split`](https://github.com/pytorch/pytorch/blob/f622ec8084c1778c6943474796efa52e9b9cf695/torch/utils/data/dataset.py#L262), but where the splits don't need necessarily to be random.